src/sage/groups/matrix_gps/linear.py

```diff
@@ -307,3 +309,49 @@ def _check_matrix(self, x, *args):
         else:
             if x.determinant() == 0:
                 raise TypeError('matrix must non-zero determinant')
+
+    def order(self):
+        """
+        Return the order of ``self``.
+
+        EXAMPLES::
+
+            sage: G = SL(3, GF(5))
+            sage: G.order()
+            372000
+
+        TESTS:
+
+        Check if :trac:`36876` is fixed::
+
+            sage: SL(1, QQ).order()
+            1
+            sage: SL(2, ZZ).cardinality()
+            +Infinity
+
+        Check if :trac:`35490` is fixed::
+
+            sage: q = 7
+            sage: FqT.<T> = GF(q)[]
+            sage: N = T^2+1
+            sage: FqTN = QuotientRing(FqT, N*FqT)
+            sage: S = SL(2, FqTN)
+            sage: S.is_finite()
+            True
+            sage: S.order()
+            117600
+        """
+        n = self.degree()
+
+        if self.base_ring().is_finite():
+            q = self.base_ring().order()
+            ord = prod(q**n - q**i for i in range(n))
+            if self._special:
+                return ord / (q-1)
```

Comment:
This should be `//`.

Reply:
Possibly yes, but it doesn't look serious to me.  Since this ticket is about to be merged, please open a new one!

Reply:
Yes but also no, rational numbers behave weirdly, e.g. `is_prime(7 / 1)` is `False`. But I will open a new PR

Reply:
@grhkm21, have you made another PR for the changes you suggested? If not then, should I create a new PR for this?